libs/core/langchain_core/messages/tool.py

```diff
@@ -1,26 +1,80 @@
-from typing import Any, List, Literal
+from typing import Any, List, Literal, Optional
 
+from langchain_core.load import Serializable
+from langchain_core.messages.ai import AIMessage
 from langchain_core.messages.base import (
     BaseMessage,
     BaseMessageChunk,
     merge_content,
 )
 from langchain_core.utils._merge import merge_dicts
+from langchain_core.utils.tools import parse_tool_calls
 
 
-class ToolMessage(BaseMessage):
-    """Message for passing the result of executing a tool back to a model."""
+class ToolCall(Serializable):
+    name: str
+    args: dict
+    id: Optional[str] = None
 
-    tool_call_id: str
-    """Tool call that this message is responding to."""
 
-    type: Literal["tool"] = "tool"
+class ToolCallsMessage(AIMessage):
+    tool_calls: Optional[List[ToolCall]] = None
+
+
+class ToolCallsMessageChunk(ToolCallsMessage, BaseMessageChunk):
+    # Ignoring mypy re-assignment here since we're overriding the value
+    # to make sure that the chunk variant can be discriminated from the
+    # non-chunk variant.
+    type: Literal["ToolCallsMessageChunk"] = "ToolCallsMessageChunk"  # type: ignore[assignment] # noqa: E501
 
     @classmethod
     def get_lc_namespace(cls) -> List[str]:
         """Get the namespace of the langchain object."""
         return ["langchain", "schema", "messages"]
 
+    def __add__(self, other: Any) -> BaseMessageChunk:  # type: ignore
+        if isinstance(other, ToolCallsMessageChunk):
```

Comment:
should we support merging with AIMessageChunk? and should AIMessageChunk support adding ToolCallsMesasgeChunk

Reply:
ah i guess if other is AIMessageChunk then calling super().__add__ will handle that, but think we still need to update AIMessageChunk.add to handle ToolCallsMessageChunk?

Reply:
Thanks, updated this. I moved the new tool calls messages to ai.py to avoid circular imports. Now AIMessage + ToolCallsMessage = ToolCallsMessage and vice-versa.